src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LoggerMessage.cs

```diff
@@ -460,7 +476,30 @@ private static LogValuesFormatter CreateLogValuesFormatter(string formatString,
                     SR.Format(SR.UnexpectedNumberOfNamedParameters, formatString, expectedNamedParameterCount, actualCount));
             }
 
-            return logValuesFormatter;
+            return (logValuesFormatter, formatParts, formats);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int Append<TValue>(Span<char> span, TValue value, string format)
+        {
+            // ReSharper disable once MergeCastWithTypeCheck
+            if (value is ISpanFormattable)
```

Comment:
```suggestion
            if (value is ISpanFormattable spanFormattable)
```

Reply:
I believe I was falling with having it JITted without boxing. It was a while ago. Will take a look at it and use the new syntax if possible.

Reply:
Yes, today the JIT can't avoid boxing when using `if (value is ISpanFormattable sf) { sf.TryFormat(...); }` but it can when doing `if (value is ISpanFormattable) { ((ISpanFormattable)value).TryFormat(...); }`; the latter case is easier to prove correct.